web3.js/src/publickey.ts

```diff
@@ -84,7 +92,18 @@ export class PublicKey extends Struct {
    * Return the base-58 representation of the public key
    */
   toBase58(): string {
-    return bs58.encode(this.toBytes());
+    const bytes = this.toBytes();
+    let record = base58StringCache;
+    bytes.forEach(word => {
+      if (record[word] === undefined) {
+        record[word] = {value: null};
+      }
+      record = record[word];
+    });
```

Comment:
The reason that I'm OK with this tree walk on every single call is because, in my testing of Explorer (eg. loading http://explorer.solana.com/block/136609835) there was still a 2x speedup overall.